pandas/core/dtypes/dtypes.py

```diff
@@ -2046,6 +2046,15 @@ def __init__(self, pyarrow_dtype: pa.DataType) -> None:
     def __repr__(self) -> str:
         return self.name
 
+    def __hash__(self) -> int:
+        # make myself hashable
+        return hash(str(self))
```

Comment:
Would this make `StringDtype('pyarrow')` equivalent to `ArrowDtype(pa.string())`?

Reply:
No, these still evaluate to not equal:

```
In [1]: import pandas as pd

In [2]: import pyarrow as pa

In [3]: pd.StringDtype(storage="pyarrow") == pd.ArrowDtype(pa.string())
Out[3]: False
```